src/Interpreters/Context.cpp

```diff
@@ -5170,69 +5170,71 @@ ReadSettings Context::getReadSettings() const
 {
     ReadSettings res;
 
-    std::string_view read_method_str = settings.local_filesystem_read_method.value;
+    auto settings_copy = getSettings();
```

Comment:
why not just
```cpp
SharedLockGuard lock(&mutex_: mutex);
```

Reply:
I actually have another question, data race is between thread in clickhouse-local that modifies global context settings after it’s initialisation and some background thread in MergeTree. This sounds not right, background threads should be started after all modifications an initialisations in global context are done. So maybe there is a bug in clickhouse-local with initialisation order